sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionStateBuilder.scala

```diff
@@ -82,8 +82,8 @@ class HiveSessionStateBuilder(session: SparkSession, parentState: Option[Session
 
     override val postHocResolutionRules: Seq[Rule[LogicalPlan]] =
       new DetectAmbiguousSelfJoin(conf) +:
-        new DetermineTableStats(session) +:
         RelationConversions(conf, catalog) +:
+        new DetermineTableStats(session) +:
```

Comment:
`DetermineTableStats` updates statistics in `HiveTableRelation` for some cases. Updated statistics will be propagated into `HadoopFsRelation` and so the `sizeInBytes` depends on it.

As you change the order of `DetermineTableStats` to after `RelationConversions`, it could change the  `sizeInBytes` of converted `HadoopFsRelation`.

That said, if the user is willing to use `fallBackToHdfsForStatsEnabled` to calculate table size, this change will make it not work as before.

Reply:
@viirya Hive tables are converted To `HiveTableScanExec` instead of `Logical Relation`(which inturn uses HadoopFSRelation) . This happens in `org.apache.spark.sql.hive.HiveStrategies.HiveTableScans` . It will not affect HadoopFSRelation I think. Let me know if I am missing some thing
Note: conversion to HiveTableScanExec happen for Hive table with any formats other than Parquet and Orc.
In case or Orc and Parquet, it happens when the flag to convert to datasource table is disabled.

Reply:
I mean previously `DetermineTableStats` will calculate table size and update in `HiveTableRelation` before `RelationConversions`.  Then in `RelationConversions`, this calculated table size will be propagated into `HadoopFsRelation` and used by `sizeInBytes`.

Now as you change the rule order, when running `RelationConversions`, even users enable `fallBackToHdfsForStatsEnabled`, the `HadoopFsRelation` won't get the table size calculated in `DetermineTableStats` (before it is run after `RelationConversions` now).

Reply:
@viirya Thanks for catching this, I think this re-order will not be useful. I will decline this pull request.

Reply:
> Updated statistics will be propagated into HadoopFsRelation and so the sizeInBytes depends on it.

Ah, I see. I missed that code flow.